src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -32,6 +33,7 @@
 
     private $propertyTypeExtractor;
     private $attributesCache = array();
+    private $extraAttributes = array();
```

Comment:
A service should not have global states like this one. Here it is worst because it creates a bug: if the `denormalize` method fails one time because of an extra attribute, all successive calls will throw an error (even if there is no extra attribute) because `$this->extraAttributes` isn't reseted to an empty array.

Here I think that it's enough to use a local variable to store the list of extra attributes. If you want to store information available in nested `denormalize` calls, you should use the `$context` parameter.